main.cpp

```diff
@@ -117,17 +143,58 @@ struct IntType
         delete value;
     }
 
-    operator int() { return *value; }
+    operator int() const { return *value; }
     
     IntType& add(int v);
     IntType& subtract(int v);
     IntType& multiply(int v);
     IntType& divide(int v);
+    
+    IntType& pow(const IntType& v);
+    IntType& pow(const FloatType& v);
+    IntType& pow(const DoubleType& v);
+    IntType& pow(int);
+    
+    IntType& powInternal(int v);
 
 private:
     int* value { nullptr };
 };
 
+// Point class
+Point::Point(float val_x, float val_y) : x(val_x), y(val_y) {}
+Point::Point(const DoubleType& val_x, const DoubleType& val_y) : x(static_cast<float>(val_x)), y(static_cast<float>(val_y)) {}
+
+Point& Point::multiply(float m)
+{
+    x *= m;
+    y *= m;
+    return *this;
+}
+
+Point& Point::multiply(const FloatType& m)
+{
+    x *= m;
+    y *= m;
+    return *this;
+}
+
+Point& Point::multiply(const DoubleType& m)
+{
+    x *= static_cast<float>(m);
+    y *= static_cast<float>(m);
+    return *this;
+}
+
+Point& Point::multiply(const IntType& m)
+{
+    x *= m;
+    y *= m;
+    return *this;
+}
```

Comment:
D.R.Y.
All of these should call `multiply(float)` internally.
pay attention to what `multiply(float)` returns, as well, when you refactor
- [ ] revise